File: 1200-minimum-absolute-difference/1200.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        ans = []
        arr.sort()

        diff = arr[1] - arr[0]
        minIndex = 0
        for i in range(1, len(arr)-1):
            if arr[i+1] - arr[i] < diff:
                diff = arr[i+1] - arr[i]
                minIndex = i

        for i in range(minIndex, len(arr)-1):
            for j in range(i+1, len(arr)):
                if arr[j] - arr[i] == diff:
                    ans.append([arr[i], arr[j]])
                    break
                if arr[j] - arr[i] > diff:
                    break

        return ans

File: 1200-minimum-absolute-difference/1200.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        vector<vector<int>> ans;
        vector<int> prefix;
        sort(arr.begin(), arr.end());

        int minimum = INT_MAX;
        for (int i = 1; i < arr.size(); ++i) {
            int val = arr[i] - arr[i - 1];
            minimum = min(minimum, val);
            prefix.push_back(val);
        }

        for (int i = 0; i < prefix.size(); ++i) {
            if (prefix[i] == minimum) {
                ans.push_back({arr[i], arr[i + 1]});
            }
        }

        return ans;
    }
};
